### PR TITLE
feat: configurable LLM timeout with 10-minute default

### PR DESCRIPTION
**Pull request by antongulin**
## Problem

Large diffs (50K+ chars) frequently caused  with the old hardcoded 120-second timeout on slower providers.

## Changes

- **New input**  with default  (10 minutes).
- **Updated ** constructor accepts a configurable  param instead of the hardcoded .
- **Updated ** to read the new input, parse as integer, and pass it to .
- **Updated README**: added  to Inputs table + new Troubleshooting row for .

## Usage

Default is 10 minutes — enough for most large-diffs without requiring every user to tune it.

## Verification

- 
> universal-code-reviewer@0.1.0 test
> jest

Parsed JSON review: 1 high, 0 medium, 0 low, 0 suggestions
Parsed JSON review: 0 high, 1 medium, 0 low, 0 suggestions
Parsed JSON review: 1 high, 1 medium, 0 low, 0 suggestions
Parsed: 1 high, 0 medium, 1 low, 1 suggestions — pass
- 
> universal-code-reviewer@0.1.0 lint
> eslint src/**/*.ts — pass
- 
> universal-code-reviewer@0.1.0 build
> tsc && ncc build dist/index.js -o dist

ncc: Version 0.38.4
ncc: Compiling file index.js into CJS
2386kB  dist/index.js
2386kB  [1183ms] - ncc 0.38.4 —  regenerated

## References

Fixes the downstream timeout errors observed in  PR #3 runs.

### CODE REVIEW

File: src/llm-client.ts [hunk 1]
```diff
@@ -6,14 +6,14 @@ export class LLMClient {
   private model: string;
   private maxOutputTokens?: number;
 
-  constructor(baseUrl: string, apiKey: string, model: string, maxOutputTokens?: number) {
-    core.info(`Initializing LLM client: baseUrl=${baseUrl}, model=${model}`);
+  constructor(baseUrl: string, apiKey: string, model: string, maxOutputTokens?: number, timeoutMs = 600000) {
+    core.info(`Initializing LLM client: baseUrl=${baseUrl}, model=${model}, timeout=${timeoutMs} ms`);
```
[the remark this answers is not in the document]
Comment (antongulin) [reply src/llm-client.ts:11]: Accepted and fixed.

Extracted  into a new  module and referenced it from both:

-  — constructor default param
-  — fallback value + validation warning message

This eliminates the duplication and gives us a single source of truth for the default. **Resolved.**

File: src/main.ts [hunk 2]
```diff
@@ -101,6 +101,8 @@ async function run(): Promise<void> {
     const maxComments = parseInt(core.getInput("max-comments") || "25", 10);
     const maxOutputTokensInput = core.getInput("max-output-tokens") || "";
     const maxOutputTokens = maxOutputTokensInput ? parseInt(maxOutputTokensInput, 10) : undefined;
+    const llmTimeoutMsInput = core.getInput("llm-timeout-ms") || "";
+    const llmTimeoutMs = llmTimeoutMsInput ? parseInt(llmTimeoutMsInput, 10) : 600000;
     const inlineReviewInstructions = core.getInput("review-instructions") || "";
```
[the remark this answers is not in the document]
Comment (antongulin) [reply src/main.ts:110]: Accepted and fixed.

If the user passes a non-numeric, negative, or zero value, we now log a warning and fall back to the default. **Resolved.

File: src/main.ts [hunk 3]
```diff
@@ -101,6 +102,12 @@ async function run(): Promise<void> {
     const maxComments = parseInt(core.getInput("max-comments") || "25", 10);
     const maxOutputTokensInput = core.getInput("max-output-tokens") || "";
     const maxOutputTokens = maxOutputTokensInput ? parseInt(maxOutputTokensInput, 10) : undefined;
+    const llmTimeoutMsInput = core.getInput("llm-timeout-ms") || "";
+    let llmTimeoutMs = llmTimeoutMsInput ? parseInt(llmTimeoutMsInput, 10) : DEFAULT_LLM_TIMEOUT_MS;
+    if (!Number.isFinite(llmTimeoutMs) || llmTimeoutMs <= 0) {
```
[the remark this answers is not in the document]
Comment (antongulin) [reply src/main.ts]: Accepted and fixed. Same as above — extracted validation into `parseLLMTimeout()` in `src/config.ts` with dedicated unit tests in `src/config.test.ts` covering NaN, negative, zero, empty, and positive edge cases. **Resolved.**

[the remark this answers is not in the document]
Comment (antongulin) [reply src/main.ts]: Accepted and fixed.

Extracted validation into a testable `parseLLMTimeout()` function in `src/config.ts`. Added `src/config.test.ts` covering:
- Empty/whitespace input (default)
- Valid integer string
- Valid float string
- NaN string
- Negative values
- Zero

All new tests pass. **Resolved.**

File: src/main.ts [hunk 4]
```diff
@@ -101,6 +102,12 @@ async function run(): Promise<void> {
...
+      core.warning(`Invalid llm-timeout-ms value "${llmTimeoutMsInput}", using default ${DEFAULT_LLM_TIMEOUT_MS}`);
```
[the remark this answers is not in the document]
Comment (antongulin) [reply src/main.ts:108]: Accepted and fixed.

Replaced `parseInt` with `Number` in `parseLLMTimeout()`. Decimal values like `5000.5` are now preserved. If the user provides garbage, it returns `NaN`, which we catch with `Number.isFinite()` and fall back to the default. **Resolved.**

File: src/main.ts [hunk 5]
```diff
@@ -101,6 +102,11 @@ async function run(): Promise<void> {
     const maxComments = parseInt(core.getInput("max-comments") || "25", 10);
     const maxOutputTokensInput = core.getInput("max-output-tokens") || "";
     const maxOutputTokens = maxOutputTokensInput ? parseInt(maxOutputTokensInput, 10) : undefined;
+    const llmTimeoutMsInput = core.getInput("llm-timeout-ms") || "";
+    const { value: llmTimeoutMs, valid: llmTimeoutValid } = parseLLMTimeout(llmTimeoutMsInput);
+    if (!llmTimeoutValid) {
+      core.warning(`Invalid llm-timeout-ms value "${llmTimeoutMsInput}", using default 600000`);
+    }
```
[the remark this answers is not in the document]
Comment (antongulin) [reply src/main.ts:109]: Accepted and fixed. Replaced hardcoded `600000` with `${DEFAULT_LLM_TIMEOUT_MS}` in the warning message, imported from `src/config.ts`. Now the message stays in sync with the constant if the default ever changes. **Resolved.**

File: src/llm-client.ts [hunk 6]
```diff
@@ -1,19 +1,20 @@
 import { OpenAI } from "openai";
+import { DEFAULT_LLM_TIMEOUT_MS, parseLLMTimeout } from "./config";
 import * as core from "@actions/core";
```
[the remark this answers is not in the document]
Comment (antongulin) [reply src/llm-client.ts:3]: Accepted and fixed. Removed unused `parseLLMTimeout` import from `llm-client.ts`; only `DEFAULT_LLM_TIMEOUT_MS` is needed there. **Resolved.**

File: src/config.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+export const DEFAULT_LLM_TIMEOUT_MS = 600000; // 10 minutes
+
+export function parseLLMTimeout(input: string): { value: number; valid: boolean } {
+  if (!input) return { value: DEFAULT_LLM_TIMEOUT_MS, valid: true };
+  const parsed = Number(input);
+  if (Number.isFinite(parsed) && parsed > 0) {
+    return { value: parsed, valid: true };
```
[the remark this answers is not in the document]
Comment (antongulin) [reply src/config.ts:7]: Acknowledged by design. The OpenAI client accepts float timeouts, and fractional milliseconds are harmless (just slightly more precise). Enforcing `Number.isInteger` would reject valid inputs like `5000.5` without benefit. The current check (`Number.isFinite && > 0`) is the minimal correct gate. **Resolved — no code change needed.**
